Source/WebInspectorUI/UserInterface/Controllers/ConsoleManager.js

```diff
@@ -39,9 +39,28 @@ WI.ConsoleManager = class ConsoleManager extends WI.Object
         this._isNewPageOrReload = false;
         this._remoteObjectsToRelease = null;
 
+        this._customLoggingChannels = [];
+
+        this._snippets = new Set;
+
+        WI.ConsoleSnippet.addEventListener(WI.SourceCode.Event.ContentDidChange, this._handleSnippetContentChanged, this);
+
         WI.Frame.addEventListener(WI.Frame.Event.MainResourceDidChange, this._mainResourceDidChange, this);
 
-        this._customLoggingChannels = [];
+        WI.Target.registerInitializationPromise((async () => {
+            let serializedSnippets = await WI.objectStores.consoleSnippets.getAll();
+
+            this._restoringSnippets = true;
```

Comment:
Should we declare this variable in the initializer itself as well to avoid mutating the shape of `ConsoleManager` in when doing this deferred work?